### PR TITLE
Handle multiple new lines

### PR DESCRIPTION
**Pull request by camillebaldock**
Creates the concept of a `MarkdownLine` and a `MarkdownBlankLine`. 
We now treat newlines as something to parse separately from a paragraph, and no longer as an ignorable delimiter. 

Introduction of Jekyll-style delimiters for front matter for ease of parsing. 

Please come talk to me, not hugely happy with the use of 'blocks' in the parsing logic, but could not figure out a way to phrase the grammar without parslet stalling...

### CODE REVIEW

File: lib/smartdown/model/block.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module Smartdown
+  module Model
+    Block = Struct.new(:elements) 
+  end
+end
```
Comment (jackscotti) [inline lib/smartdown/model/block.rb]: A new line please :)

**Comment by camilleldn** [comment]
Code Review pair: 
* renaming of blocks as Elements
* stop using MarkdownBlankLine, revert to just MarkdownLine

File: lib/smartdown/model/node.rb [hunk 2]
```diff
@@ -4,7 +4,14 @@ module Smartdown
   module Model
     Node = Struct.new(:name, :elements, :front_matter) do
       def initialize(name, elements, front_matter = nil)
-        super(name, elements, front_matter || Smartdown::Model::FrontMatter.new)
+        all_elements = elements.map do |element|
```
Comment (dsingleton) [inline lib/smartdown/model/node.rb]: I'd be good to have an inline comment here (and/or in the `Elements` class itself) explaining why there is the `Elements` model that needs flattening. We don't want too much information about how the parser works to leak into other layers, but we should explain it somewhere. What do you think?

File: lib/smartdown/model/node.rb [hunk 3]
```diff
@@ -73,8 +80,10 @@ def elements_of_kind(*kinds)
       def as_markdown(block)
         case block
         when Smartdown::Model::Element::MarkdownHeading
-          "# #{block.content}\n"
-        when Smartdown::Model::Element::MarkdownParagraph
+          "# #{block.content}"
+        when Smartdown::Model::Element::MarkdownLine
+          block.content
+        when Smartdown::Model::Element::MarkdownBlankLine
```
Comment (dsingleton) [inline lib/smartdown/model/node.rb]: `MarkdownBlankLine` is gone now right?

**Comment by dsingleton** [comment]
Looks good to me, pending a couple of comments i'd be happy to merge :+1:

**Pull request merged**